vispy/gloo/tests/test_texture.py

```diff
@@ -714,4 +715,18 @@ def test_texture_3D_internalformats():
     _test_texture_internalformats(Texture3D, (10, 10, 10))
 
 
+@requires_pyopengl()
+@pytest.mark.parametrize('input_dtype', [np.uint8, np.uint16, np.float32, np.float64])
+@pytest.mark.parametrize('output_dtype', [np.uint8, np.uint16, np.float32, np.float64])
+def test_texture_set_data_different_dtype(input_dtype, output_dtype):
+    data2D = np.random.rand(20, 20).astype(input_dtype)
+    tex2D = Texture2D(data2D)
+    tex2D[:10] = np.array(1, dtype=output_dtype)
+    tex2D.set_data(data2D.astype(output_dtype))
+
+    data3D = np.random.rand(20, 20, 20).astype(input_dtype)
+    tex3D = Texture3D(data3D)
+    tex2D[:10] = np.array(1, dtype=output_dtype)
+    tex3D.set_data(data3D.astype(output_dtype))
```

Comment:
Is this test just checking that a certain data type can be ingested by the Texture? Is there something we should be checking after it is set to confirm validity or is the fact that it doesn't raise an exception good enough?

Otherwise, the 2D versus 3D could/should be made into another pytest parameter and then an `if` statement to handle if a Texture2D or Texture3D should be made.

Also, typo, `tex2D[:10]` in the 3D block of code.

Reply:
This is just testing that no error arises. The `image` and `volume` tests make sure that the rendering goes as expected as well, since `Texture` does not provide a way to get back the data from the GPU afaik. Updated the test!